StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp8/Lightup/SwitchExpressionArmSyntaxWrapperTests.cs

```diff
@@ -0,0 +1,118 @@
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+    using StyleCop.Analyzers.Lightup;
+    using Xunit;
+
+    public class SwitchExpressionArmSyntaxWrapperTests
+    {
+        [Fact]
+        public void TestNull()
+        {
+            var syntaxNode = default(SyntaxNode);
+            var switchExpressionSyntax = (SwitchExpressionArmSyntaxWrapper)syntaxNode;
+            Assert.Null(switchExpressionSyntax.SyntaxNode);
+            Assert.Throws<NullReferenceException>(() => switchExpressionSyntax.Pattern);
+            Assert.Throws<NullReferenceException>(() => switchExpressionSyntax.WhenClause);
+            Assert.Throws<NullReferenceException>(() => switchExpressionSyntax.Expression);
+            Assert.Throws<NullReferenceException>(() => switchExpressionSyntax.EqualsGreaterThanToken);
+
+            var patternSyntax = SyntaxFactory.ConstantPattern(SyntaxFactory.LiteralExpression(SyntaxKind.NullLiteralExpression));
+            Assert.Throws<NullReferenceException>(() => switchExpressionSyntax.WithPattern((PatternSyntaxWrapper)patternSyntax));
+
+            var whenClause = SyntaxFactory.WhenClause(SyntaxFactory.LiteralExpression(SyntaxKind.TrueLiteralExpression));
+            Assert.Throws<NullReferenceException>(() => switchExpressionSyntax.WithWhenClause((WhenClauseSyntaxWrapper)whenClause));
+
+            Assert.Throws<NullReferenceException>(() => switchExpressionSyntax.WithExpression(SyntaxFactory.LiteralExpression(SyntaxKind.NullLiteralExpression)));
+            Assert.Throws<NullReferenceException>(() => switchExpressionSyntax.WithEqualsGreaterThanToken(SyntaxFactory.Token(SyntaxKind.EqualsGreaterThanToken)));
+        }
+
+        [Fact]
+        public void TestProperties()
+        {
+            var syntaxNode = this.CreateSwitchExpressionArm();
+
+            Assert.True(syntaxNode.IsKind(SyntaxKind.SwitchExpressionArm));
+            Assert.True(syntaxNode.IsKind(SyntaxKindEx.SwitchExpressionArm));
+
+            var wrapper = (SwitchExpressionArmSyntaxWrapper)syntaxNode;
+            Assert.Same(syntaxNode, wrapper.SyntaxNode);
+            Assert.Same(syntaxNode.Pattern, wrapper.Pattern.SyntaxNode);
+            Assert.Same(syntaxNode.WhenClause, wrapper.WhenClause.SyntaxNode);
+            Assert.Same(syntaxNode.Expression, wrapper.Expression);
+            Assert.Equal(syntaxNode.EqualsGreaterThanToken, wrapper.EqualsGreaterThanToken);
+
+            // Pattern
+            var newPattern = SyntaxFactory.ConstantPattern(SyntaxFactory.LiteralExpression(SyntaxKind.NullLiteralExpression));
+            var wrapperWithModifiedPattern = wrapper.WithPattern((PatternSyntaxWrapper)newPattern);
+            Assert.NotNull(wrapperWithModifiedPattern.SyntaxNode);
+            Assert.NotSame(syntaxNode.Pattern, wrapperWithModifiedPattern.Pattern.SyntaxNode);
+            Assert.Equal(SyntaxKind.ConstantPattern, wrapperWithModifiedPattern.Pattern.SyntaxNode.Kind());
```

Comment:
💡 You should be able to also assert that the new expression arm has the correct pattern. I don't remember if `Assert.Same` would work here, but I believe there is least an `IsEquivalentTo` for syntax nodes that could be used to compare `wrapperWithModifiedPattern.Pattern` and `newPattern`.